src/callbacks/solution_saving.jl

```diff
@@ -209,99 +209,61 @@ function (finalize::SolutionSavingCallback)(c, u, t, integrator)
     return nothing
 end
 
-function Base.show(io::IO, cb::DiscreteCallback{<:Any, <:SolutionSavingCallback})
-    @nospecialize cb # reduce precompilation time
-
-    solution_saving = cb.affect!
-    print(io, "SolutionSavingCallback(interval=", solution_saving.interval, ")")
-end
-
-function Base.show(io::IO,
-                   cb::DiscreteCallback{<:Any,
-                                        <:PeriodicCallbackAffect{<:SolutionSavingCallback}})
-    @nospecialize cb # reduce precompilation time
-
-    solution_saving = cb.affect!.affect!
-    print(io, "SolutionSavingCallback(dt=", solution_saving.interval, ")")
-end
-
-function Base.show(io::IO,
-                   cb::DiscreteCallback{<:Any, <:Any, <:Any, <:SolutionSavingCallback})
-    @nospecialize cb # reduce precompilation time
-
-    solution_saving = cb.finalize
-    print(io, "SolutionSavingCallback(save_times=", solution_saving.save_times, ")")
-end
-
-function Base.show(io::IO, ::MIME"text/plain",
-                   cb::DiscreteCallback{<:Any, <:SolutionSavingCallback})
-    @nospecialize cb # reduce precompilation time
-
-    if get(io, :compact, false)
-        show(io, cb)
+function Base.show(io::IO, cb::DiscreteCallback)
+    solution_saving = get_solution_saving_callback(cb)
+    if solution_saving !== nothing
+        if !isempty(solution_saving.save_times)
+            print(io, "SolutionSavingCallback(save_times=", solution_saving.save_times, ")")
+        elseif isa(solution_saving.interval, Integer)
+            print(io, "SolutionSavingCallback(interval=", solution_saving.interval, ")")
+        else
+            print(io, "SolutionSavingCallback(dt=", solution_saving.interval, ")")
+        end
     else
-        solution_saving = cb.affect!
-        cq = collect(solution_saving.custom_quantities)
-
-        setup = [
-            "interval" => solution_saving.interval,
-            "custom quantities" => isempty(cq) ? nothing : cq,
-            "save initial solution" => solution_saving.save_initial_solution ?
-                                       "yes" : "no",
-            "save final solution" => solution_saving.save_final_solution ? "yes" :
-                                     "no",
-            "output directory" => abspath(solution_saving.output_directory),
-            "prefix" => solution_saving.prefix
-        ]
-        summary_box(io, "SolutionSavingCallback", setup)
+        # Fallback to default show method for AbstractCallback
+        invoke(Base.show, Tuple{IO, AbstractCallback}, io, cb)
```

Comment:
What if this is dispatched elsewhere? Then you overwrite this to instead use the generic function for `AbstractCallback`. Even if we don't do this in TrixiParticles, just loading TrixiParticles with another package that dispatches on a subtype of `DiscreteCallback` would break this other package.